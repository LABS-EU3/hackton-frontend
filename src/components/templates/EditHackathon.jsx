import React from "react";
import HackathonForm from "./HackathonForm";
import { useSelector } from "react-redux";
import { useParams } from "react-router-dom";

const EditHackathon = () => {
  const { id } = useParams();
  const event = useSelector(state =>
    state.events.data.find(ev => ev.id === Number(id))
  );

  return <HackathonForm initialState={event} />;
};

export default EditHackathon;
