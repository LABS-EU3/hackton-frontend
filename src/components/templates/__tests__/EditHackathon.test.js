import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import EditHackathon from '../EditHackathon';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render <EditHackathon/> template correctly", () => {
  
    it("renders the User EditHackathon templatet correctly", () => {
      const template = () => renderWithRouter(EditHackathon)
      expect(template).toMatchSnapshot();
    });
  });