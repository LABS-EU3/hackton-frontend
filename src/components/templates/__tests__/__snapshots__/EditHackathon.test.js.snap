// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render <EditHackathon/> template correctly renders the User EditHackathon templatet correctly 1`] = `[Function]`;
